packages/blade/src/components/Checkbox/_decisions/decisions.md

```diff
@@ -0,0 +1,220 @@
+      defaultValue={['anurag', 'chaitanya']}
+    >
+      <Checkbox value="anurag">Anurag</Checkbox>
+      <Checkbox value="kamlesh">Kamlesh</Checkbox>
+      <Checkbox value="chaitanya">Chaitanya</Checkbox>
+    </CheckboxGroup>
+  );
+};
+```
+
+### Note on CheckboxGroup
+
+For properties `isDisabled`, `validationState` we want the CheckboxGroup to pass down these props to all the Checkboxes inside of it.
+
+For showing optional/required label in `CheckboxGroup` we will introduce a prop `necessityIndicator` which will only render a `(optional)/(required)` tag in the CheckboxGroupLabel,
+And the consumer will provide proper validation logic and depending on their use case and requirements they will pass `validationState` in conjunction with `necessityIndicator` and they should also use `helpText` to convey the correct intent and extra information to the user.
+
+<img width="578" alt="Screenshot of checkbox with optional tag" src="https://user-images.githubusercontent.com/35374649/178486036-9371917f-b599-4f8b-b455-92d2c5f91c4b.png">
+
+
+**Example of user managed validation**
+
+Example of only one checkbox needs to be checked:
+
+```tsx
+function AnyOneHasToBeSelected() {
+  const [selected, setSelected] = React.useState([]);
+
+  return (
+    <CheckboxGroup
+      label="Do you love any of the fruits?"
+      helpText="You can select multiple"
+      value={selected}
+      onChange={setSelected}
+      validationState={selected.length === 0 ? 'error' : undefined}
+    >
+      <Checkbox value="apple">apple</Checkbox>
+      <Checkbox value="mango">mango</Checkbox>
+      <Checkbox value="orange">orange</Checkbox>
+      <Checkbox value="banana">banana</Checkbox>
+    </CheckboxGroup>
+  );
+}
+```
+
+Example of every checkboxes needs to be checked:
+
+```tsx
+function AllOfThemHasToBeSelected() {
+  const [selected, setSelected] = React.useState([]);
+
+  return (
+    <CheckboxGroup
+      label="Do you agree to all the terms?"
+      helpText="Select all"
+      value={selected}
+      onChange={setSelected}
+      hasError={selected.length < 3 ? 'error' : undefined}
+    >
+      <Checkbox value="use-ts">I will use TS</Checkbox>
+      <Checkbox value="no-any">I won't use any</Checkbox>
+      <Checkbox value="use-js">I won't use JS</Checkbox>
+    </CheckboxGroup>
+  );
+}
+```
+
+## Accessibility
+
+- Checkbox - https://www.w3.org/WAI/ARIA/apg/patterns/checkbox/
+- CheckboxGroup - The group's accessibility is bit tricky to get right, thus I'll implement and refine it and then document it here. 
+
+## Open Questions
+
+1. Currently we only have `helpText` in Checkbox & CheckboxGroup and when we set `hasError` it becomes the `negative` variant. The problem is the content of the `helpText` should also change in certain scenarios to convey the exact error. 
+
+With this approach users will have to handle the logic
+
+```tsx
+const [hasError, setInvalid] = React.useState(false);
+
+<Checkbox 
+  hasError={hasError}
```

Comment:
a bit confused. Don't see this prop in API reference chart above. Are we use `validationState: 'error'` or `hasError` 🤔 ?

Reply:
validationState. Will fix it.